src/grib2.h.in

```diff
@@ -21,6 +21,12 @@
 
 #cmakedefine G2_PNG_ENABLED 1 /**< Decoding/encoding PNG is enabled */
 #cmakedefine G2_JPEG2000_ENABLED 1 /**< Decoding/encoding JPEG2000 is enabled */
+#ifndef G2_PNG_ENABLED
+#define G2_PNG_ENABLED 0
+#endif
+#ifndef G2_JPEG2000_ENABLED
+#define G2_JPEG2000_ENABLED 0
+#endif
```

Comment:
OK, this is not the way it is supposed to work. ;-)

The cmakedefine pragma will cause G2_PNG_ENABLED to be defined to 1 if PNG is enabled, and not defined if not.

So in your code you can do:

```
#ifdef G2_PNG_ENABLED
/* some code here */
#endif

```

So this PR is not needed...

Reply:
OK.  That is good to know and makes sense.  I will handle this accordingly in grib2io's Cython wrapper.

Reply:
Sure enough, this situation can easily be accommodated for in Cython

```cython
cdef extern from "grib2.h":
    """
    #ifndef G2_PNG_ENABLED
        #define G2_PNG_ENABLED 0
    #endif
    #ifndef G2_JPEG2000_ENABLED
        #define G2_JPEG2000_ENABLED 0
    #endif
    """
    cdef char *G2_VERSION
    cdef int G2_PNG_ENABLED
    cdef int G2_JPEG2000_ENABLED
```

You can add extra definitions as docstrings in pyx files.  With the above, `G2_PNG_ENABLED` and `G2_JPEG2000_ENABLED` can be defined and set to 0 when **_not_** is defined in `grib2.h`.

Reply:
So you you want to close this PR?

Reply:
Yes, this PR can be closed.